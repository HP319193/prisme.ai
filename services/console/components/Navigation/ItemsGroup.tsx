import { PlusCircleOutlined } from '@ant-design/icons';
import { StretchContent } from '@prisme.ai/design-system';
import { Tooltip } from 'antd';
import { FC } from 'react';
import ChevronIcon from '../../icons/chevron.svgr';

export interface ItemsGroupProps {
  title: string;
  onClick?: () => void;
  onAdd?: () => void;
  tooltip?: string;
  open: boolean;
}
export const ItemsGroup: FC<ItemsGroupProps> = ({
  title,
  open,
  onClick,
  onAdd,
  tooltip = '',
  children,
}) => {
  return (
    <div className="flex flex-1 leading-[2.5rem]">
      <div className="flex flex-1 flex-col max-w-full">
        <div className="flex flex-1 flex-row items-center border-b-[1px]">
          <button
            className="flex flex-1 flex-row items-center outline-none focus:outline-none p-4"
            onClick={onClick}
          >
            <Tooltip title={title} placement="left">
              <div className="flex m-2 mr-4 w-[1.6rem] h-[1.6rem] justify-center">
                <ChevronIcon
                  width="1rem"
                  className={` transition-transform ${
                    open ? '' : '-rotate-90'
                  }`}
                />
              </div>
            </Tooltip>
            <div className="flex flex-1 font-bold whitespace-nowrap">
              {title}
            </div>
          </button>
          {onAdd && (
            <Tooltip title={tooltip} placement="left">
              <button
                className="flex outline-none focus:outline-none p-4 hover:text-accent"
                onClick={onAdd}
              >
                <PlusCircleOutlined />
              </button>
            </Tooltip>
          )}
        </div>
        <div className="flex flex-1">
          <StretchContent visible={open} className="whitespace-nowrap flex-1">
            {children}
          </StretchContent>
        </div>
      </div>
    </div>
  );
};

export default ItemsGroup;
