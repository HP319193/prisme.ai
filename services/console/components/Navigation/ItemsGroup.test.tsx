import renderer from 'react-test-renderer';
import ItemsGroup from './ItemsGroup';

it('should render', () => {
  const root = renderer.create(
    <ItemsGroup open title="Foo">
      <div>Bar</div>
    </ItemsGroup>
  );
  expect(root.toJSON()).toMatchSnapshot();
});

it('should render closed', () => {
  const root = renderer.create(
    <ItemsGroup open={false} title="Foo">
      <div>Bar</div>
    </ItemsGroup>
  );
  expect(root.toJSON()).toMatchSnapshot();
});
