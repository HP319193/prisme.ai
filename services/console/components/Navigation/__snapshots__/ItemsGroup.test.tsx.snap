// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="flex flex-1 leading-[2.5rem]"
>
  <div
    className="flex flex-1 flex-col max-w-full"
  >
    <div
      className="flex flex-1 flex-row items-center border-b-[1px]"
    >
      <button
        className="flex flex-1 flex-row items-center outline-none focus:outline-none p-4"
      >
        <div
          className="flex m-2 mr-4 w-[1.6rem] h-[1.6rem] justify-center"
        >
          <SvgrURL
            className=" transition-transform "
            width="1rem"
          />
        </div>
        <div
          className="flex flex-1 font-bold"
        >
          Foo
        </div>
      </button>
    </div>
    <div
      className="flex flex-1"
    >
      <div>
        Bar
      </div>
    </div>
  </div>
</div>
`;

exports[`should render closed 1`] = `
<div
  className="flex flex-1 leading-[2.5rem]"
>
  <div
    className="flex flex-1 flex-col max-w-full"
  >
    <div
      className="flex flex-1 flex-row items-center border-b-[1px]"
    >
      <button
        className="flex flex-1 flex-row items-center outline-none focus:outline-none p-4"
      >
        <div
          className="flex m-2 mr-4 w-[1.6rem] h-[1.6rem] justify-center"
        >
          <SvgrURL
            className=" transition-transform -rotate-90"
            width="1rem"
          />
        </div>
        <div
          className="flex flex-1 font-bold"
        >
          Foo
        </div>
      </button>
    </div>
    <div
      className="flex flex-1"
    >
      <div>
        Bar
      </div>
    </div>
  </div>
</div>
`;
